src/frontend/js/widgets/UserLogin/index.tsx

```diff
@@ -75,23 +75,13 @@ const UserLogin = ({ profileUrls = {} }: UserLoginProps) => {
    */
   const { user, destroy, login, register } = useSession();
   const intl = useIntl();
-  const {
-    item: joanieUserProfile,
-    states: { isFetched: isJoanieUserProfileFetched },
-  } = useJoanieUserProfile();
-
-  const isAnonymous = user === null;
-  const isUserFetched = user !== undefined;
-  const isReady = isAnonymous || (isUserFetched && isJoanieUserProfileFetched);
+  const joanieUserAbilities = useJoanieUserAbilities();
 
   const userMenuItems = useMemo(() => {
-    if (isReady) {
+    if (joanieUserAbilities !== null) {
       const unauthorizedUrlFilter = (urlItem: UserMenuItem) => {
         if (urlItem.key === 'dashboard_teacher') {
-          return AbilitiesHelper.can(
-            joanieUserProfile,
-            abilityActions.CAN_ACCESS_TEACHER_DASHBOARD,
-          );
+          return joanieUserAbilities!.can(abilityActions.ACCESS_TEACHER_DASHBOARD);
```

Comment:
With the check above `!` is useless I think.

```suggestion
          return joanieUserAbilities.can(abilityActions.ACCESS_TEACHER_DASHBOARD);
```